### ISSUE TITLE
Batch update products by SKU

### ISSUE THREAD

**Issue by tiansial** [woocommerce/wc-api-php#159]
Is there any way to batch update by given array of sku?

Just like this:

`$data = [
        'update' => [
            [
                'sku' => '2148503000610',
                'regular_price' => 50
            ]
        ]
    ];

    $woocommerce->post('products/batch', $data);`

**Comment by jasie** [comment]
The docs only speak of id (http://woocommerce.github.io/woocommerce-rest-api-docs/?php#batch-update-products). I assume, you tested with sky instead of id, and it failed?
I think, this can't be supported, as it is not save that the sku is unique. It seems the same sku can be assigned to different products (see https://github.com/woocommerce/woocommerce/issues/5712).
So the id is the ONLY truly save unique identifier for a product.

Since every product has an id: is it unknown to you or can't you get it in order to use the batch update via id, like so:
wc_get_product_id_by_sku('your_sku')

**Comment by tiansial** [comment]
Yeah, I'm not able to update by SKU and I guess it's because of that. It can really be duplicate.

I built a function using wc_get_product_id_by_sku, but I'm afraid of the performance hit, getting 9000's SKU's before batch updating with the API

```
function BuildWooArray($array) {
  $productsToUpdate = array();

  foreach($array as $value) {
      $productsToUpdate[] = array(
          'id' => wc_get_product_id_by_sku($value['codigoprod']),
          'regular_price' => $value['preco'],
          'status' => $disponivel = ($value['disponivel'] == 1 ? 'publish' : 'draft')
      );
  }

  return $productsToUpdate;
}

$dataToUpdate = [
   'update' => BuildWooArray($RCSoftData)
];

$woocommerce->post('products/batch', $dataToUpdate);
```

**Comment by strarsis** [comment]
@tiansial: I tried this once (for https://github.com/woocommerce/wc-api-php/issues/157) by adding an API extension for WooCommerce API (which is using the WordPress REST API) - but I wasn't able to get it running.
But adding your own custom endpoint would allow running the bottleneck code on the server instead.

**Comment by jasie** [comment]
@tiansial yes, if you do this every time (e.g. on page load), this is too expensive. BUT you could do the get ID by SKU just ONCE and store the result in your database or in a json file.
Of course, when you access that data for the batch update, you have to check, if you have a complete sku-id-pair. Whenever an ID is missing, you fetch just the IDs, that are missing and add those to your "lookup table". Would that work for you?

**Comment by tiansial** [comment]
@jasie, I'm not running get ID by SKU on all products on every page load. I'm integrating WooCommerce with a company's ERP to keep all products synced (price, stock, etc).

I'm thinking building a cron job running the above function to update every product update.
On the other hand, on the product single page and checkout, I'm requesting the ERP API to get the last stock and price. That way the user is always updated.

That's so far my solution anyway.
